app/helpers/cases_helper.rb

```diff
@@ -189,6 +189,15 @@ def action_links_for_allowed_events(kase, *events)
     end
   end
 
+  def action_buttons_for_allowed_events(kase, *events)
+    allowed_events = events.find_all do |event_name|
+      policy(kase).__send__ "#{event_name}?"
+    end
+    allowed_events.map do |event_name|
+      __send__ "action_button_for_#{event_name}", kase
+    end
+  end  
+
```

Comment:
Can rewrite the above code to be more 'ruby' style:

```
events
  .find_all { |name| policy(kase).send("#{name}?") } 
  .map  { |name| send("action_button_for_#{name}", kase) }
```

Is this new code? If so should have a corresponding test ideally.

Reply:
Used this slick version instead :) 
Used existing test, but let me know if we need a more generic test, rather than testing the button that's generated.